File: backend/api/models/Space/invitedUsers.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const invitedUserSchema = Schema({
    by: {
        type: String
    },
    roles: {
        type: String,
        enum: ["collaborator", "viewer"]
    },
    spaceId: {
        type: Schema.Types.ObjectId,
        ref: "spaces"
    },
    addedBy: {
        type: Schema.Types.ObjectId,
        ref: "users"
    },
    status: {
        type: String,
        enum: ["pending", "joined", "rejected"]
    },
    active: {
        type: Boolean,
        default: true
    },
}, {
    timestamps: true
})

module.exports = mongoose.model("invitedUsers", invitedUserSchema);

File: frontend/lib/screens/spaces/models/invited_users.dart
import 'package:json_annotation/json_annotation.dart';

part 'invited_users.g.dart';

enum Role { admin, collaborator, viewer }

extension RoleX on Role {
  bool get isAdmin => this == Role.admin;
}

enum InvitedUserStatus { pending, joined, rejected }

@JsonSerializable()
class InvitedUsers {
  const InvitedUsers({
    required this.id,
    required this.by,
    required this.role,
    required this.spaceId,
    required this.addedBy,
    this.status = InvitedUserStatus.pending,
    this.active = false,
  });

  @JsonKey(name: '_id')
  final String id;

  final String by;

  @JsonKey(name: 'roles')
  final Role role;

  final String spaceId;
  final String addedBy;

  @JsonKey(defaultValue: InvitedUserStatus.pending)
  final InvitedUserStatus status;

  @JsonKey(defaultValue: true)
  final bool active;

  factory InvitedUsers.fromJson(Map<String, dynamic> json) =>
      _$InvitedUsersFromJson(json);

  Map<String, dynamic> toJson() => _$InvitedUsersToJson(this);
}
